### PR TITLE
[TAS-67] Main page

### PR DESCRIPTION
**Pull request by weebao**
https://github.com/cics-web-dev/Chem-Web-App/assets/85828374/adb4ac37-1ac3-44a3-9907-447ac8b6f2f8

### CODE REVIEW

File: frontend/src/routes/+page.svelte [hunk 1]
```diff
@@ -1,2 +1,194 @@
...
+            class="relative rounded-xl bg-neutral-900 px-8 py-2 text-2xl font-semibold text-white opacity-0 transition-all duration-300 hover:cursor-pointer hover:bg-neutral-700 dark:bg-slate-200 dark:text-slate-800 dark:hover:bg-slate-400"
+            on:focus={() => goto('/login')}
+        >
+            Start learning
+        </button>
+    </div>
+    <ul id="hover-card" class="relative flex w-96 flex-col">
+        <li class="group relative flex aspect-square w-1/4 items-center justify-center self-center hover:cursor-pointer">
+            <div
+                class="absolute top-0 -z-10 aspect-square w-full rotate-45 rounded-lg bg-slate-200 dark:bg-slate-400 transition-[width,top] group-hover:top-6 group-hover:z-10 group-hover:w-[225%]"
```
Comment (algebra2boy) [inline frontend/src/routes/+page.svelte:127]: There are a lot of duplicated css properties here in line 127, 141, 152, 166 such as `absolute -z-10 aspect-square w-full rotate-45 rounded-lg bg-slate-200 dark:bg-slate-400 group-hover:top-6 group-hover:z-10 group-hover:w-[225%]`. I'd suggest to declare and use a common css class, that way it is easier for the future reader understands what is going on.

File: frontend/src/routes/+page.svelte [hunk 2]
```diff
@@ -1,2 +1,194 @@
...
+
+    /* ---- ANIMATION FOR PARTICLES ---- */
+    const randomSpringCoords = () => {
+        const xMin = 10;
+        const xMax = 40;
+        const yMin = 0;
+        const yMax = 90;
+        const coords = spring({
+            x: Math.random() * (xMax - xMin) + xMin,
+            y: Math.random() * (yMax - yMin) + yMin
```
Comment (algebra2boy) [inline frontend/src/routes/+page.svelte:81]: Can you write a helper function for this? Line 87 and 88 uses the same logic.

**Review by algebra2boy — CHANGES_REQUESTED** [review]
The addition looks amazing. Just a few minor requested change. I was curious what it looks like on mobile, and I think it would be best to also handle the layout on mobile as well.
Screenshot 2024-05-10 at 1 04 43 PM

File: frontend/src/routes/+page.svelte [hunk 3]
```diff
@@ -1,2 +1,194 @@
...
+    import Molecule from '$icons/Main/Molecule.svelte';
+    import RoundBeaker from '$icons/Main/RoundBeaker.svelte';
+    import ArrowDown from '$icons/Sidebar/ArrowDown.svelte';
+
+    // If the user is already logged in, redirect them to questions (for now)
+    $: {
+        if ($page.data.user) {
+            goto('/question/1');
+        }
+    }
```
[the remark this answers is not in the document]
Comment (algebra2boy) [reply frontend/src/routes/+page.svelte]: I have addressed this in one of commit in this PR.

[the remark this answers is not in the document]
Comment (weebao) [reply frontend/src/routes/+page.svelte]: Got it thank you so much Yongye

File: frontend/src/routes/+page.svelte [hunk 4]
```diff
@@ -1,2 +1,194 @@
...
+    const animationDuration = 0.3;
+    let animationStarted = false;
+    let timeline: gsap.core.Timeline;
+
+    $: {
+        if (animateTextWrapper) {
+            Array.from(animateTextWrapper.children).forEach((elem) => {
+                timeline = gsap.timeline();
+                const splitText = new SplitType(elem as HTMLElement, {
+                    types: 'lines,words,chars'
```
[the remark this answers is not in the document]
Comment (weebao) [reply frontend/src/routes/+page.svelte:30]: Yeah you're right this is probably not necessary

File: frontend/src/routes/+page.svelte (hunk 1 above)
Comment (weebao) [reply frontend/src/routes/+page.svelte:118]: Got it will change that later

File: frontend/src/routes/+page.svelte (hunk 2 above)
Comment (weebao) [reply frontend/src/routes/+page.svelte:72]: Yep will do

**Comment by weebao** [comment]
> @weebao Can you try to resolve the merge conflict here? We can work on the responsive issue in another PR. Once you resolve it, I can merge it right away.

I believe I resolved it. Can you please check Yongye?

**Review by algebra2boy — APPROVED** [review]

**Pull request merged**